src/gl/value.js

```diff
@@ -28,920 +27,434 @@ export interface Value<T> {
-            const gl = this.context.gl;
-            gl.bindTexture(gl.TEXTURE_2D, v);
-            this.current = v;
-            this.dirty = false;
-        }
+    _set(v: ?WebGLTexture) {
+        const gl = this.gl;
+        gl.bindTexture(gl.TEXTURE_2D, v);
     }
 }
 
-export class BindVertexBuffer implements Value<?WebGLBuffer> {
-    context: Context;
-    current: ?WebGLBuffer;
-    default: ?WebGLBuffer;
-    dirty: boolean;
-
-    constructor(context: Context) {
-        this.context = context;
-        this.default = null;
-        this.current = this.default;
-        this.dirty = false;
+export class BindVertexBuffer extends BaseValue<?WebGLBuffer> {
+    getDefault(): WebGLBuffer {
+        return null;
     }
-
-    get(): ?WebGLBuffer { return this.current; }
-
-    setDefault(): void { this.set(this.default); }
-
-    set(v: ?WebGLBuffer): void {
-        if (this.current !== v || this.dirty === true) {
-            const gl = this.context.gl;
-            gl.bindBuffer(gl.ARRAY_BUFFER, v);
-            this.current = v;
-            this.dirty = false;
-        }
+    _set(v: ?WebGLBuffer) {
+        const gl = this.gl;
+        gl.bindBuffer(gl.ARRAY_BUFFER, v);
     }
 }
 
-export class BindElementBuffer implements Value<?WebGLBuffer> {
-    context: Context;
-    current: ?WebGLBuffer;
-    default: ?WebGLBuffer;
-    dirty: boolean;
-
-    constructor(context: Context) {
-        this.context = context;
-        this.default = null;
-        this.current = this.default;
-        this.dirty = false;
+export class BindElementBuffer extends BaseValue<?WebGLBuffer> {
+    getDefault(): WebGLBuffer {
+        return null;
     }
-
-    get(): ?WebGLBuffer { return this.current; }
-
-    setDefault(): void { this.set(this.default); }
-
-    set(v: ?WebGLBuffer): void {
+    set(v: ?WebGLBuffer) {
         // Always rebind
-        const gl = this.context.gl;
+        const gl = this.gl;
         gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, v);
         this.current = v;
         this.dirty = false;
     }
 }
 
-export class BindVertexArrayOES implements Value<any> {
-    context: Context;
-    current: any;
-    default: any;
-    dirty: boolean;
+export class BindVertexArrayOES extends BaseValue<any> {
+    vao: any;
 
     constructor(context: Context) {
-        this.context = context;
-        this.default = null;
-        this.current = this.default;
-        this.dirty = false;
+        super(context);
+        this.vao = context.extVertexArrayObject;
     }
-
-    get(): any { return this.current; }
-
-    setDefault(): void { this.set(this.default); }
-
-    set(v: any): void {
-        if (this.current !== v && this.context.extVertexArrayObject || this.dirty === true) {
-            this.context.extVertexArrayObject.bindVertexArrayOES(v);
-            this.current = v;
-            this.dirty = false;
+    getDefault(): any {
+        return null;
+    }
+    _set(v: any) {
+        if (this.vao) {
```

Comment:
Thinking out loud to document the change here...

Before, if we didn't have `context.extVertexArrayObject`, we would rely on code in `context.js` to never set `this.dirty` to `true`, so we'd never try to bind the object. Now, we don't depend on that code in `context.js` to keep us from attempting to bind, but it still prevents us from storing any value in `this.current` (although it also looks like all the calls to `bindVertexArrayOES.set` also check for the extension before running).

So looks like just one more layer of belt-and-suspenders protection against accidentally using a missing extension. :+1:.